src/Program.cs

```diff
@@ -32,13 +32,14 @@ private static async Task<int> Main(string[] args)
                 .AddOption(new Option(new[] { "-w", "--workspace" }, Resources.The_solution_or_project_file_to_operate_on_If_a_file_is_not_specified_the_command_will_search_the_current_directory_for_one, new Argument<string>(() => null)))
                 .AddOption(new Option(new[] { "-v", "--verbosity" }, Resources.Set_the_verbosity_level_Allowed_values_are_quiet_minimal_normal_detailed_and_diagnostic, new Argument<string>() { Arity = ArgumentArity.ExactlyOne }.FromAmong(_verbosityLevels)))
                 .AddOption(new Option(new[] { "--dry-run" }, Resources.Format_files_but_do_not_save_changes_to_disk, new Argument<bool>()))
+                .AddOption(new Option(new[] { "--check" }, Resources.Terminate_with_non_zero_exit_code_if_any_files_formatted, new Argument<bool>()))
                 .UseVersionOption()
                 .Build();
 
             return await parser.InvokeAsync(args).ConfigureAwait(false);
         }
 
-        public static async Task<int> Run(string workspace, string verbosity, bool dryRun, IConsole console = null)
+        public static async Task<int> Run(string workspace, string verbosity, bool dryRun, bool check, IConsole console = null)

```

Comment:
💡 Rather than pass this argument, we could _always_ simply return the number of files formatted from this method. The caller could then determine what to do with that number.

Reply:
💭 Might need to use negative codes for errors, and positive codes for successful finding of number of files that need to be formatted. Otherwise the caller would not know whether 1 means error or no-error when `--check` is not given by the user.

Reply:
> 💡 Rather than pass this argument, we could _always_ simply return the number of files formatted from this method. The caller could then determine what to do with that number.

Currently as I understand this method is being called by `CommandLineBuilder` command in code above:

`new Command("dotnet-format", 
handler: CommandHandler.Create(typeof(Program).GetMethod(nameof(Run))))`

New '--check' option is being passed in the same manner as all other options. So currently its not clear for me how to make the change you described, could you please advice?...

Reply:
> 💭 Might need to use negative codes for errors, and positive codes for successful finding of number of files that need to be formatted. Otherwise the caller would not know whether 1 means error or no-error when `--check` is not given by the user.

Should we make some changes in this PR?..

Reply:
@roman-petrov I think the current implementation is probably what is expected. If we need to make a change, we can do it later.